cfme_perf.sh refactoring

dmetzger57 opened the pull request:
Besides refactoring, added upgrading pip to resolve build issue with he default older pip. Tested on RHEL7 and Centos7

amavinag committed:
Add pytest_generator which add scenario name to test rather index.

  * added pytest_generator to idele tests.

dmetzger57 committed:
cfme_perf.sh refacoring and add upgrading pip to resolve build issues
with the older version.

dmetzger57 committed:
Removed installation of package group development-tools, not needed.

akrzos commented on cfme_perf.sh:
```diff
@@ -7,12 +7,36 @@ else
+                     freetype-devel \
+                     openssl-devel
+
+echo "### Creating Virtual Environment: .cfme_perf"
```
We should add .cfme_perf to the .gitignore for one less item to appear in git status and prevent virtualenv files from sneaking into a commit.

akrzos commented:
Worked on fedora24.  Should we squash the commits and merge it?

dmetzger57 commented:
just tried it on Fedora 24 Server and it needed the numpy package to succeed without error for me. Let me do a little more testing.

akrzos commented:
Interesting.  numpy is in the python requirements.txt already, was there a numpy rpm you are referencing?

dmetzger57 commented:
I didn't add a reference, least not the I recall :-)

<SNIP>
  UPDATING build/lib.linux-x86_64-2.7/matplotlib/_version.py
  set build/lib.linux-x86_64-2.7/matplotlib/_version.py to '1.5.1'
  running build_ext
  building 'matplotlib.ft2font' extension
  setupext.py:819: UserWarning: The C headers for numpy could not be found.
You may need to install the development package
-  "The C headers for numpy could not be found. "*
  creating build/temp.linux-x86_64-2.7
  creating build/temp.linux-x86_64-2.7/src
  gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall
  -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions
  -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches
  -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic
  -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall
  -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions
  -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches
  -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic
  -D_GNU_SOURCE -fPIC -fwrapv -fPIC
  -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib_ft2font_ARRAY_API
  -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION
  -I/tmp/pip-build-_18Wpt/matplotlib/.eggs/numpy-1.11.1-py2.7-linux-x86_64.egg/numpy/core/include
  -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/local/include
  -I/usr/include -I. -I/usr/include/python2.7 -c src/ft2font.cpp -o
  build/temp.linux-x86_64-2.7/src/ft2font.o
  cc1plus: error:
  /tmp/pip-build-_18Wpt/matplotlib/.eggs/numpy-1.11.1-py2.7-linux-x86_64.egg/numpy/core/include:
  Not a directory
  error: command 'gcc' failed with exit status 1
  
  ---
  
  _Failed building wheel for matplotlib_
  <SNIP>

# 

Dennis
Why is it that all of the instruments seeking intelligent life in the
universe are pointed away from Earth?

On Mon, Aug 22, 2016 at 11:47 AM, Alex Krzos notifications@github.com
wrote:

> Interesting. numpy is in the python requirements.txt already, was there a
> numpy rpm you are referencing?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/akrzos/cfme-performance/pull/55#issuecomment-241456715,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ALtde9CmzLzonNvH5p5GIUxdCr2i7ZKpks5qicSbgaJpZM4Jo00N
> .

dmetzger57 committed:
Add numpy to installation packages

There was a build failure do to missing numpy headers, this resolves that.

dmetzger57 commented:
Removed the WIP, installs (setup) clean on Fedora Server / CentOS / RHEL

# 

Dennis
Why is it that all of the instruments seeking intelligent life in the
universe are pointed away from Earth?

On Mon, Aug 22, 2016 at 12:33 PM, Dennis Metzger dennis.metzger@gmail.com
wrote:

> I didn't add a reference, least not the I recall :-)
> 
> <SNIP>
>   UPDATING build/lib.linux-x86_64-2.7/matplotlib/_version.py
>   set build/lib.linux-x86_64-2.7/matplotlib/_version.py to '1.5.1'
>   running build_ext
>   building 'matplotlib.ft2font' extension
>   setupext.py:819: UserWarning: The C headers for numpy could not be
> found. You may need to install the development package
> -  "The C headers for numpy could not be found. "*
>   creating build/temp.linux-x86_64-2.7
>   creating build/temp.linux-x86_64-2.7/src
>   gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall
>   -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions
>   -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches
>   -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic
>   -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall
>   -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions
>   -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches
>   -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic
>   -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib_ft2font_ARRAY_API
>   -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION -I/tmp/pip-build-_18Wpt/
>   matplotlib/.eggs/numpy-1.11.1-py2.7-linux-x86_64.egg/numpy/core/include
>   -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/local/include
>   -I/usr/include -I. -I/usr/include/python2.7 -c src/ft2font.cpp -o
>   build/temp.linux-x86_64-2.7/src/ft2font.o
>   cc1plus: error: /tmp/pip-build-_18Wpt/matplotlib/.eggs/numpy-1.11.1-
>   py2.7-linux-x86_64.egg/numpy/core/include: Not a directory
>   error: command 'gcc' failed with exit status 1
>   
>   ---
>   
>   _Failed building wheel for matplotlib_
>   <SNIP>
> 
> # 
> 
> Dennis
> Why is it that all of the instruments seeking intelligent life in the
> universe are pointed away from Earth?
> 
> On Mon, Aug 22, 2016 at 11:47 AM, Alex Krzos notifications@github.com
> wrote:
> 
> > Interesting. numpy is in the python requirements.txt already, was there a
> > numpy rpm you are referencing?
> > 
> > —
> > You are receiving this because you authored the thread.
> > Reply to this email directly, view it on GitHub
> > https://github.com/akrzos/cfme-performance/pull/55#issuecomment-241456715,
> > or mute the thread
> > https://github.com/notifications/unsubscribe-auth/ALtde9CmzLzonNvH5p5GIUxdCr2i7ZKpks5qicSbgaJpZM4Jo00N
> > .

akrzos committed:
Updates to the single page UI workload.

Including two scenarios by default with the config file. (explorer, show_list)

akrzos commented on cfme_perf.sh:
```diff
@@ -7,12 +7,37 @@ else
 . ./.cfme_perf/bin/activate
+
+echo "### Upgrading pip"
+pip install --upgrade pip
```
For rhel we should have easy_install pip before this right?  (I assume centos would need that as well?)

arcolife commented:
@akrzos looks like I didn't look at this PR before submitting https://github.com/akrzos/cfme-performance/pull/56 
err..

akrzos committed:
Add isMaster to version Ansible role

Also run create appliances playbvook from localhost. (Some sort of ansible issue with using 127.0.0.1)

akrzos committed:
Add is57 for postdeploy

+ Also add isMaster for uiworker patch

akrzos committed:
Allow get/set vmdb yaml on 57 appliances.
